plugins/grid-bookmark/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { lazy } from 'react'
+
+import BookmarkIcon from '@material-ui/icons/Bookmark'
+import BookmarksIcon from '@material-ui/icons/Bookmarks'
+
+import WidgetType from '@jbrowse/core/pluggableElementTypes/WidgetType'
+import Plugin from '@jbrowse/core/Plugin'
+import PluginManager from '@jbrowse/core/PluginManager'
+import { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'
+import { getSession } from '@jbrowse/core/util'
+import ViewType from '@jbrowse/core/pluggableElementTypes/ViewType'
+import { isSessionModelWithWidgets } from '@jbrowse/core/util'
+
+import {
+  stateModelFactory as GridBookmarkStateModelFactory,
+  configSchema as GridBookmarkConfigSchema,
+} from './GridBookmarkWidget'
+
+export default class extends Plugin {
+  name = 'GridBookmarkPlugin'
+
+  install(pluginManager: PluginManager) {
+    pluginManager.addWidgetType(() => {
+      return new WidgetType({
+        name: 'GridBookmarkWidget',
+        heading: 'Bookmarked regions',
+        configSchema: GridBookmarkConfigSchema,
+        stateModel: GridBookmarkStateModelFactory(pluginManager),
+        ReactComponent: lazy(
+          () => import('./GridBookmarkWidget/components/GridBookmarkWidget'),
+        ),
+      })
+    })
+
+    pluginManager.extendPluggable('LinearGenomeView', LinearGenomeView => {
+      const { stateModel } = LinearGenomeView as ViewType
+      const newStateModel = stateModel.extend((self: LinearGenomeViewModel) => {
+        const superMenuItems = self.menuItems
+        const superRubberBandMenuItems = self.rubberBandMenuItems
+        return {
+          actions: {
+            activateBookmarkWidget() {
+              const session = getSession(self)
+              if (isSessionModelWithWidgets(session)) {
+                // @ts-ignore
+                let bookmarkWidget = session.widgets.get('GridBookmark')
```

Comment:
Seems like this ts-ignore could potentially by removed by fixing the type of `session.widgets`, which I think is not correct. I think it should be:
```diff
diff --git a/packages/core/util/types/index.ts b/packages/core/util/types/index.ts
--- a/packages/core/util/types/index.ts
+++ b/packages/core/util/types/index.ts
@@ -119,7 +119,7 @@ export function isSessionModelWithConfigEditing(
 /** abstract interface for a session that manages widgets */
 export interface SessionWithWidgets extends AbstractSessionModel {
   visibleWidget?: { id: string }
-  widgets?: unknown[]
+  widgets: Map<string, unknown>
   addWidget(
     typeName: string,
     id: string,
```

Reply:
I can try that